### PR TITLE
Context lookup API endpoint part 1

### PR DESCRIPTION
**Pull request by jkppr**
**This PR is part one of working on issue #2439 to create a context lookup for event fields!** 

This PR includes:

- A configuration file `context_links.yaml` that is used to provide the context link configuration.
- Adjustments in existing configurations to load this file.
- An API endpoint resource `contextlinks.py` that loads and validates the config file.
- Adjusted routes to make the API endpoint available.

### CODE REVIEW

File: timesketch/api/v1/resources/contextlinks.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            else:
+                check_ok = False
+
+            # Verify that redirect_warning is defined and type bool
+            if not isinstance(item_dict.get("redirect_warning"), bool):
+                check_ok = False
+
+            if not check_ok:
+                logger.warning(
+                    "Failed to load context link: '{0:s}'! Please check the "
```
Comment (jaegeral) [inline timesketch/api/v1/resources/contextlinks.py:94]: Couldn't we provide more info on what is missing instead?

File: timesketch/api/v1/resources/contextlinks.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        """Handles GET request to the resource.
+        Returns:
+            JSON object including version info
+        """
+        context_link_config = load_yaml_config("CONTEXT_LINKS_CONFIG_PATH")
+
+        response = {}
+
+        if not context_link_config:
+            logger.warning("The config file 'data/context_links.yaml' is empty!")
```
[the remark this answers is not in the document]
Comment (jkppr) [reply timesketch/api/v1/resources/contextlinks.py:44]: Good point. However, after our talk about what to report in the logs and what not, I will opt for removing the warning. Since the default is an empty config file, this would create unnecessary log entries for a feature that the operator maybe just does not want to use. 
TL;DR: An empty config file is working as intended and hence does not need a warning.

File: timesketch/api/v1/resources/contextlinks.py (hunk 1 above)
Comment (jkppr) [reply timesketch/api/v1/resources/contextlinks.py:94]: Based on our discussion I refactored the input verification to be
1. better readable in code
2. bundled all failures in one message at the end to prevent spamming of the log file (1 message independent of how many entries are bad)
3. each error message now contains more granular info on which part of the config is bad

File: timesketch/api/v1/resources/contextlinks.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+        response = {}
+
+        if not context_link_config:
+            return jsonify(response)
+
+        verification_results = []
+
+        # Check mandatory fields are correctly defined.
+        for item in context_link_config:
```
Comment (berggren) [inline timesketch/api/v1/resources/contextlinks.py:49]: Why we are validating the YAML file every time the API endpoint executes? This implies that we don't trust the supplied, server side, YAML file to follow the needed format.

Instead, we could add a validation function to tsctl that the server admin can run to validate the yaml file before it is used in the system. Then it is up to the admin to check that it has a valid file. We could then use something like jsonschema to validate the fields to simplify this logic.

WDYT? Let's discuss this in person.

**Review by berggren — CHANGES_REQUESTED** [review]

File: timesketch/api/v1/resources/contextlinks.py (hunk 3 above)
Comment (jkppr) [reply timesketch/api/v1/resources/contextlinks.py]: Thanks for the input. As discussed:

- Moved validation to tsctl for manual execution by an admin.
- Switched to jsonschema validation
- Removed validation from the API endpoint itself

File: timesketch/tsctl.py [hunk 4]
```diff
@@ -21,6 +21,7 @@
 import click
 from flask.cli import FlaskGroup
 from sqlalchemy.exc import IntegrityError
+from jsonschema import validate, ValidationError
```
Comment (berggren) [inline timesketch/tsctl.py:24]: Do we need to add this to requirements.txt?

**Review by berggren — CHANGES_REQUESTED** [review]
Looks good, one question we need to address before merge

File: timesketch/tsctl.py (hunk 4 above)
Comment (jkppr) [reply timesketch/tsctl.py:24]: Added as discussed.

**Review by berggren — APPROVED** [review]
